ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1452,7 +1452,13 @@ extension SignalType {
 					var state = state
 					state.pendingValue = value
 
-					let proposedScheduleDate = state.previousDate?.dateByAddingTimeInterval(interval) ?? scheduler.currentDate
+					let proposedScheduleDate: NSDate
+					if let previousDate = state.previousDate where previousDate.compare(scheduler.currentDate) == .OrderedAscending {
```

Comment:
I think this deserves a comment explaining why `previousDate` could be at or after `scheduler.currentDate`.